### PR TITLE
imath 3.1.0

### PR DESCRIPTION
**Pull request by SMillerDev**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
Error looks related to https://github.com/emscripten-core/emscripten-fastcomp-clang/issues/29. Probs ARM-specific.

**Comment by carlocab** [comment]
> Probs ARM-specific.

Intel nodes are fine. Linux has a different problem, though:

```
==> brew linkage --test openexr
==> FAILED
Missing libraries:
  unexpected (libImath-3_0.so.28)
```

CC @Homebrew/linux

**Comment by carlocab** [comment]
`vips` and `opencv` need rev bumps. Would probably be good to just do that along with my changes from #81242.

**Comment by SMillerDev** [comment]
@carlocab you want to bump imath in that PR?

**Comment by carlocab** [comment]
Could do that. We should figure out solutions to the ARM and Linux issues first though, as we don't want the changes there to be waiting indefinitely for fixes to those too.

**Comment by SMillerDev** [comment]
Isn't openexr just linked to the old library for imath on Linux? Seems pretty simple to me.

The ARM fix will be more involved I guess.

**Comment by carlocab** [comment]
Yea, I guess Linux can be fixed with a rev bump. Didn’t register as the linkage failure message looks different compared to macOS linkage errors 😄

**Comment by carlocab** [comment]
ARM failure reported at AcademySoftwareFoundation/Imath#183.

**Comment by carlocab** [comment]
ARM failure might be fixable with
```
ENV.append "CPPFLAGS", "-DIMATH_HALF_NO_TABLES_AT_ALL" if Hardware::CPU.arm?
```

File: Formula/imath.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@ class Imath < Formula
   depends_on "cmake" => :build
 
   def install
+    ENV.append "CPPFLAGS", "-DIMATH_HALF_NO_TABLES_AT_ALL" if Hardware::CPU.arm?
+
```
Comment (carlocab) [inline Formula/imath.rb:20]:
```suggestion

```

**Comment by carlocab** [comment]
ARM fix didn't work, sadly. Guess we'll have to wait to hear from upstream.

**Comment by cary-ilm** [comment]
FWIW,  "IMATH_HALF_NO_TABLES_AT_ALL" was an earlier name for the option during development, it has been renamed to "IMATH_HALF_NO_LOOKUP_TABLE" in th 3.1 release. But that won't address the failure, which stems from:

#    if __has_include(<x86intrin.h>)
#        include <x86intrin.h>

which needs an extra conditional that I'm not certain about, but one other other project members will, shortly. Thanks.

**Comment by SMillerDev** [comment]
Builds are fine now, @carlocab do you want to use your vips fixes here?

**Comment by carlocab** [comment]
Yea, I'll cherry-pick them onto your PR branch.

**Comment by carlocab** [comment]
I missed the `vips` revision bump. I somehow thought that was covered by the revision bump to `visp`. ...

**Review by carlocab — DISMISSED** [review]
This LGTM -- just remove the `automerge-skip` label when you're happy with it.

**Comment by carlocab** [comment]
Ah, merge conflict. Wonderful.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**